library/std/src/sys/pal/windows/thread_local_key.rs

```diff
@@ -313,17 +319,25 @@ unsafe fn run_dtors() {
         // Use acquire ordering to observe key initialization.
         let mut cur = DTORS.load(Acquire);
         while !cur.is_null() {
-            let key = (*cur).key.load(Relaxed) - 1;
+            let pre_key = (*cur).key.load(Relaxed);
```

Comment:
Is it really worth making this `Relaxed`? If we make it `Acquire` then we're guaranteed to fully sync with the key's `init()` in case initialization completed, which seems much safer.

Reply:
With the new change, it has to be `Acquire` because we might observe key initialization through this load. Previously, `Relaxed` was fine since all registered `StaticKey`s in the list already had their key initialized.

Reply:
> we might observe key initialization through this load

Are you sure? There's still a release-acquire pair via registering the dtor and the picking up the dtor here, so `TlsAlloc` happens-before `TlsGetValue`.

Reply:
Yeah, right. Anyway, the `Acquire` here is fine.